scripts/quests/outlands/An_Undying_Pledge.lua

```diff
@@ -0,0 +1,132 @@
+-----------------------------------
+-- Log ID: 5, Quest ID: 149
+-- Stray Cloud : !pos -18 1 -20 252
+-- qm5         : !pos 133 -9 221 176
+-----------------------------------
+local ssgID = zones[xi.zone.SEA_SERPENT_GROTTO]
+
+local quest = Quest:new(xi.questLog.OUTLANDS, xi.quest.id.outlands.AN_UNDYING_PLEDGE)
+
+quest.reward =
+{
+    item = xi.item.LIGHT_BUCKLER,
+    fameArea = xi.fameArea.NORG,
+    fame = 50,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.NORG) >= 4
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Stray_Cloud'] = quest:progressEvent(225),
+
+            onEventFinish =
+            {
+                [225] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Stray_Cloud'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        return quest:event(228)
+                    elseif quest:getVar(player, 'Prog') == 1 then
+                        return quest:event(229)
+                    elseif quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(227)
+                    end
+                end,
+            },
+
+            onTriggerAreaEnter =
+            {
+                [1] = function(player, triggerArea)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        return quest:progressEvent(226)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [226] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [227] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.CALIGINOUS_BLADE)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.SEA_SERPENT_GROTTO] =
+        {
+            ['qm5'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 1 and
+                        npcUtil.popFromQM(player, npc, ssgID.mob.GLYRYVILU, { claim = true, hide = 0 })
+                    then
+                        return quest:messageSpecial(ssgID.text.BODY_NUMB_DREAD)
+                    elseif
+                        quest:getVar(player, 'Prog') == 2
+                    then
+                        return quest:progressEvent(18)
+                    end
```

Comment:
```lua
                    local questProgress = quest:getVar(player, 'Prog')
                    if
                        questProgress == 1 and
                        npcUtil.popFromQM(player, npc, ssgID.mob.GLYRYVILU, { claim = true, hide = 0 })
                    then
                        return quest:messageSpecial(ssgID.text.BODY_NUMB_DREAD)
                    elseif questProgress == 2 then
                        return quest:progressEvent(18)
                    end
```
To expand:
- As claywar said, whenever you have to check the same thing more than once, store it on a variable first so we dont repeat operations.
- When there is only one condition, theres no need to align the `if/elseif` separate from the condition or the `then`